Hamlib 4.2

In hamlib 4.2 the FILPATHLEN was replaced by HAMLIB_FILPATHLEN. Please fix line 47 of RigThread.cpp

Hi,

This is still breaking macOS builds from brew (see https://github.com/BatchDrake/suscan/runs/4291017085?check_suite_focus=true):

```
/usr/local/Homebrew/Library/Homebrew/shims/mac/super/clang++ -DUSE_HAMLIB -D__MACOSX_CORE__ -DaudioSupport_EXPORTS -I/tmp/soapyaudio-20211122-30114-1vl4vfk/SoapyAudio-soapy-audio-0.1.1 -I/usr/local/Cellar/rt-audio/5.1.0_1/include/rtaudio -std=c++11 -Wc++11-extensions -O3 -DNDEBUG -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk -fPIC -fvisibility=hidden -fvisibility-inlines-hidden   -pthread -Wall -Wextra -Wnon-virtual-dtor -MD -MT CMakeFiles/audioSupport.dir/Version.cpp.o -MF CMakeFiles/audioSupport.dir/Version.cpp.o.d -o CMakeFiles/audioSupport.dir/Version.cpp.o -c /tmp/soapyaudio-20211122-30114-1vl4vfk/SoapyAudio-soapy-audio-0.1.1/build/Version.cpp
/tmp/soapyaudio-20211122-30114-1vl4vfk/SoapyAudio-soapy-audio-0.1.1/RigThread.cpp:41:56: error: use of undeclared identifier 'FILPATHLEN'
        strncpy(rig->state.rigport.pathname, rigFile.c_str(), FILPATHLEN - 1);
                                                              ^
1 error generated.
```

Are there any plans to merge this fix back to master soon?

Thanks in advance,

--- a/Changelog.txt
+++ b/Changelog.txt
@@ -1,3 +1,9 @@
+Release 0.1.2 (pending)
+==========================
+
+- Fix build for Hamlib 4.2 and up
+- Fix hamlib control for Softrock
+
 Release 0.1.1 (2019-05-12)
 ==========================
 

--- a/RigThread.cpp
+++ b/RigThread.cpp
@@ -44,7 +44,11 @@ void RigThread::run() {
         return;
     }
 
-	strncpy(rig->state.rigport.pathname, rigFile.c_str(), FILPATHLEN - 1);
+    //provide HAMLIB_FILPATHLEN when building against Hamlib < 4.2
+    #ifndef HAMLIB_FILPATHLEN
+    #define HAMLIB_FILPATHLEN FILPATHLEN
+    #endif
+	strncpy(rig->state.rigport.pathname, rigFile.c_str(), HAMLIB_FILPATHLEN - 1);
     if (serialRate > 0)  // bypass for rate 0 used for USB controlled radios
 	    rig->state.rigport.parm.serial.rate = serialRate;
 

